navigation component

24sint opened the pull request:
navigation component added in the component and pattern library

#### Background Information
One of the molecule in the page layout.


#### Impact
We can access the navigation component on the aside.



#### Issue
Solves: [Molecule] Build a page navigation component #87
<img width="790" alt="Screen Shot 2019-08-10 at 6 38 11 PM" src="https://user-images.githubusercontent.com/30203095/62827660-aa221580-bba1-11e9-9f0c-14c98167e2b2.png">

MustafaAlsaeedi commented on components/Molecule/navigation-component.html:
```diff
@@ -0,0 +1,72 @@
+          </nav>
+
+     <div class="navigation-components">
+
```
You probably don't need these lines (39-46) as you are creating just one molecule.

24sint replied:
@MustafaAlsaeedi Thank you so much for your feedback, that is very helpful and knowledgeable. You guys teach us what does mean code review. On the other hand, I added line 39-46 to show there will be the main banner. And if you see the Casa Marianella layout the aside navigation component display when you click the **About** from the main banner menu. So, I used the property and value **display: none** and then I added the click add an event listener using JavaScript.

MustafaAlsaeedi commented on components/Molecule/navigation-component.html:
```diff
@@ -0,0 +1,72 @@
+aside{
+    display: none;
+    background-color: whitesmoke;
+    float: right;
```
We are using grid so you can get red of float. Also, it will be display grid later after we add this molecule to the page layout, so you don't need display: none. There is no background color in the original task #87, do you just like to have one?

MustafaAlsaeedi commented on components/Molecule/navigation-component.html:
```diff
@@ -0,0 +1,72 @@
+    display: none;
+    background-color: whitesmoke;
+    float: right;
+    width: 25.5%;
```
width will be defined in fraction (ex: 1fr) in grid layout.

MustafaAlsaeedi commented on components/Molecule/navigation-component.html:
```diff
@@ -0,0 +1,72 @@
+}
+ nav{
+     width: 100%;
+ }
```
No need for width line 22

MustafaAlsaeedi commented:
Great job it looks great after removing the comment I mentioned in my reviews here is how it looks on my end. 
![Screen Shot 2019-08-10 at 9 21 30 PM](https://user-images.githubusercontent.com/29615921/62828458-dd21d480-bbb4-11e9-9204-b4d1ca588654.png)

24sint commented:
@MustafaAlsaeedi Thank you so much for your feedback, that is very helpful and knowledgeable. You guys teach us what does mean code review. I checked all of your comments, and I fixed that. I like the last edited code page layout. Only one of the feedback I added line 39-46, to show there will be the main banner. And if you see the Casa Marianella layout the aside navigation component display when you click the **About** from the main banner menu. So, that I added the click add an event listener using JavaScript.

24sint commented:
@MustafaAlsaeedi I appreciate your time, effort, and knowledge. He clearly defines how grids work and how to make our code very easy, clear, and readable. We have done more and also fixed my **navigation-component PR.** Thanks again.

MustafaAlsaeedi commented:
Great job Sintu! You many not need `navigation-component` as you are creating just a molecule that will be used inside a grid layout. But, you can have it for clarity.
